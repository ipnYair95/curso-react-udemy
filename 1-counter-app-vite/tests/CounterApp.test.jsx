const { render, screen, fireEvent } = require("@testing-library/react");
const { CounterApp } = require("../src/CounterApp");

describe('CounterApp ', () => {

    const value = 100;

    test('debe hacer match con snapshort', () => {

        const { container } = render(<CounterApp value={value} />);

        expect(screen.getByText(100)).toBeTruthy();
        expect(container).toMatchSnapshot();

    });

    test('debe de incrementar con el btn +1', () => {

        render(<CounterApp value={value} />);

        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('101')).toBeTruthy();

    });

    test('debe de decrementar con el btn -1', () => {

        render(<CounterApp value={value} />);

        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('99')).toBeTruthy();

    });

    test('debe de funcionar el btn reset', () => {

        render(<CounterApp value={value} />)

        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        //fireEvent.click( screen.getByText('Reset') );
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        expect(screen.getByText(value)).toBeTruthy();

    });


});