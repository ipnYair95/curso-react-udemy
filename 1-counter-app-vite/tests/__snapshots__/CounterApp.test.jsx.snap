// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp  debe hacer match con snapshort 1`] = `
<div>
  <h1>
     Counter App 
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
  <button>
     +1 
  </button>
</div>
`;
